Lib/inspect.py

```diff
@@ -3199,10 +3201,10 @@ def _bind(self, args, kwargs, *, partial=False):
                         break
                     elif param.name in kwargs:
                         if param.kind == _POSITIONAL_ONLY:
-                            msg = '{arg!r} parameter is positional only, ' \
-                                  'but was passed as a keyword'
-                            msg = msg.format(arg=param.name)
-                            raise TypeError(msg) from None
+                            # Raise a TypeError once we are sure there is no
+                            # **kwargs param later.
+                            pos_only_param_in_kwargs.append(param)
+                            continue
```

Comment:
Removing these lines will also produce a TypeError (with different, but correct error message). So why not use a simpler code?

Reply:
Ah, so you tries to get the same error message as for calling the function. Well, it makes sense.